frontend/src/layout/navigation-3000/components/TopBar.tsx

```diff
@@ -110,30 +106,31 @@ export function TopBar(): JSX.Element | null {
 }
 
 interface BreadcrumbProps {
-    breadcrumb: FinalizedBreadcrumb
-    index: number
+    breadcrumb: IBreadcrumb
     here?: boolean
 }
 
-function Breadcrumb({ breadcrumb, index, here }: BreadcrumbProps): JSX.Element {
+function Breadcrumb({ breadcrumb, here }: BreadcrumbProps): JSX.Element {
     const { renameState } = useValues(breadcrumbsLogic)
     const { tentativelyRename, finishRenaming } = useActions(breadcrumbsLogic)
     const [popoverShown, setPopoverShown] = useState(false)
 
+    const joinedKey = Array.isArray(breadcrumb.key) ? breadcrumb.key.map(String).join(':') : String(breadcrumb.key)
```

Comment:
Nit: Pulling this out to a selector function (or just a util) solves it for both Breadcrumb and Here and can then be used in the other comment I mentioned about the fragment key

Reply:
Yup. When it comes to DRY, I think three repetitions are a good rule of thumb usually, so with the fragment in the game now, a util is definitely best